app/controllers/api/v1/npq_profiles_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class NpqProfilesController < Api::ApiController
+      include ApiTokenAuthenticatable
+      include JSONAPI::ActsAsResourceController
```

Comment:
controller action all dealt with by `jsonapi-resources` out of the box. which is why there is `def create`

Reply:
I guess it adds a bunch of other actions, right? But hopefully they can't be accessed cause we control the routes?

Reply:
at routing level you can disable any routes you don't want like a normal rails `resources` with `only` see routes file